tests/ragger/app/client.py

```diff
@@ -1,17 +1,50 @@
 from enum import IntEnum, auto
-from typing import Iterator, Dict, List
+from typing import Optional
 from ragger.backend import BackendInterface
 from ragger.utils import RAPDU
-from ethereum_client.command_builder import EthereumCmdBuilder
-from ethereum_client.setting import SettingType, SettingImpl
-from ethereum_client.eip712 import EIP712FieldType
-from ethereum_client.response_parser import EthereumRespParser
+from ragger.navigator import NavInsID, NavIns, NanoNavigator
+from app.command_builder import EthereumCmdBuilder
+from app.setting import SettingType, SettingImpl
+from app.eip712 import EIP712FieldType
+from app.response_parser import EthereumRespParser
 import signal
 import time
+from pathlib import Path
+import keychain
+
+
+ROOT_SCREENSHOT_PATH = Path(__file__).parent.parent
+
+class   DOMAIN_NAME_TAG(IntEnum):
+    STRUCTURE_TYPE = 0x01
+    STRUCTURE_VERSION = 0x02
+    CHALLENGE = 0x12
+    SIGNER_KEY_ID = 0x13
+    SIGNER_ALGO = 0x14
+    SIGNATURE = 0x15
+    DOMAIN_NAME = 0x20
+    COIN_TYPE = 0x21
+    ADDRESS = 0x22
+
+def format_tlv(tag: int, value) -> bytes:
+    if isinstance(value, int):
+        value = value.to_bytes((value.bit_length() + 7) // 8, 'big')
+    elif isinstance(value, str):
+        value = value.encode()
+
+    if not isinstance(value, bytes):
+        print("Unhandled TLV formatting for type : %s" % (type(value)))
+        return None
+
+    tlv = bytearray()
+    tlv.append(tag)
+    tlv.append(len(value))
+    tlv += value
+    return tlv
```

Comment:
Maybe create a `app/tlv.py` file for tlv helper functions

Reply:
Done. :+1: